### ISSUE TITLE
add test the check if get_factory function cares for factory_id

### ISSUE THREAD

**Issue by defiantnerd** [robbert-vdh/clap-validator#4]
A test should be added if  

```c++
auto factory = (_pluginEntry->get_factory(CLAP_PLUGIN_FACTORY_ID);
auto otherfactory = (_pluginEntry->get_factory("this is not a factory id and should return nullptr");
```

does check the factory ID and returns a nullptr if the ID is not a known factory identifier.
A plugin can be checked against bugs as such:

https://github.com/baconpaul/clap-saw-demo/blob/680798f454941d383fcd3f8e24d86fa93a62cffb/src/clap-saw-demo-pluginentry.cpp#L56

**Comment by robbert-vdh** [comment]
Great idea, I've added it to the list of tests that still need to be implemented in #1. I'll add it in a bit.

**Comment by robbert-vdh** [comment]
Implemented in https://github.com/robbert-vdh/clapval/commit/66185a6deb4fea07dd12790e96c47c6612ce5b58.
